src/vivarium/interface/cli.py

```diff
@@ -129,21 +128,25 @@ def run(
     results_root.mkdir(parents=True, exist_ok=False)
 
     configure_logging_to_file(output_directory=results_root)
-    shutil.copy(model_specification, results_root / "model_specification.yaml")
 
     output_data = {"results_directory": str(results_root)}
     input_data = {}
     if artifact_path:
         input_data["artifact_path"] = artifact_path
     override_configuration = {"output_data": output_data, "input_data": input_data}
 
-    main = handle_exceptions(run_simulation, logger, with_debugger)
-    finished_sim = main(model_specification, configuration=override_configuration)
+    sim = SimulationContext(
+        model_specification=model_specification, configuration=override_configuration
+    )
+    with open(f"{results_directory}/model_specification.yaml", "w") as f:
```

Comment:
This is fine, but if we want to systematize how we do running of simulations across our 3 types of simulations (i.e. `SimulationContext`, `InteractiveContext`, and `ParallelSimulationContext` or whatever we call it), it might make sense to figure out what the operations are across the three and how to encapsulate them to reduce duplication of code and improve flexibility. 

It may be easier to think through this once you start working on the `ParallelSimulationContext`.

Reply:
Yeah I agree. I'll merge this for now and it'll likely change later